aries_vcx/src/handlers/issuance/issuer/state_machine.rs

```diff
@@ -264,48 +222,38 @@ impl IssuerSM {
-                    (IssuerFullState::ProposalReceived(ProposalReceivedState::new(proposal, None, None, None)), thread_id)
-                }
-                _ => {
-                    warn!("Unable to process received message in this state");
-                    (IssuerFullState::Initial(state_data), thread_id)
+                    (IssuerFullState::ProposalReceived(ProposalReceivedState::new(proposal, None)), thread_id)
                 }
-            }
-            IssuerFullState::OfferSet(state_data) => match cim {
-                CredentialIssuanceMessage::CredentialOfferSend(comment) => {
-                    let cred_offer = libindy_issuer_create_credential_offer(&state_data.offer_info.cred_def_id)?;
+                CredentialIssuanceMessage::CredentialOfferSend(offer_info, comment) => {
+                    let cred_offer = libindy_issuer_create_credential_offer(&offer_info.cred_def_id)?;
                     let cred_offer_msg = CredentialOffer::create()
                         .set_id(&thread_id)
                         .set_offers_attach(&cred_offer)?
                         .set_comment(comment);
-                    let cred_offer_msg = _append_credential_preview(cred_offer_msg, &state_data.offer_info.credential_json)?;
+                    let cred_offer_msg = _append_credential_preview(cred_offer_msg, &offer_info.credential_json)?;
                     send_message.ok_or(
                         VcxError::from_msg(VcxErrorKind::InvalidState, "Attempted to call undefined send_message callback")
                     )?(&cred_offer_msg.to_a2a_message())?;
-                    (IssuerFullState::OfferSent((state_data, cred_offer, cred_offer_msg.id).into()), thread_id)
+                    (IssuerFullState::OfferSent((offer_info, cred_offer).into()), thread_id)
                 }
                 _ => {
                     warn!("Unable to process received message in this state");
-                    (IssuerFullState::OfferSet(state_data), thread_id)
+                    (IssuerFullState::Initial(state_data), thread_id)
                 }
             }
             IssuerFullState::ProposalReceived(state_data) => match cim {
-                CredentialIssuanceMessage::CredentialOfferSend(comment) => {
-                    let (cred_def_id, credential_json) = match state_data.offer_info {
-                        None => return Err(VcxError::from_msg(VcxErrorKind::InvalidState, "`set_offer()` must be called before sending credential offer after receiving proposal")),
-                        Some(offer_info) => (offer_info.cred_def_id, offer_info.credential_json)
-                    };
-                    let cred_offer = libindy_issuer_create_credential_offer(&cred_def_id)?;
+                CredentialIssuanceMessage::CredentialOfferSend(offer_info, comment) => {
+                    let cred_offer = libindy_issuer_create_credential_offer(&offer_info.cred_def_id)?;
```

Comment:
Maybe in separate PR, i'd suggest this line, and line `248` a layer up, so `CredentialOfferSend` would actually be responsible for passing in `CredentialOffer` - hence remove the dependency on `libindy/vdr-tools` here and talk in pure aries terms (eg. "please pass me valid CredentialOffer message, I don't care if it's based on libindy")

Reply:
Agree, let's do it (in a separate PR, though)